docs/source/model/notebooks/signature_examples.ipynb

```diff
@@ -0,0 +1,1001 @@
+      "The data is of type: <class 'list'>.\n",
+      "The inferred signature is:\n",
+      "\n",
+      "inputs: \n",
+      "  ['a': Array(string) (required), 'b': string (optional), 'c': {d: string (required), e: string (optional)} (required)]\n",
+      "outputs: \n",
+      "  None\n",
+      "params: \n",
+      "  None\n",
+      "\n",
+      "\n"
+     ]
+    }
+   ],
+   "source": [
+    "data = [{\"a\": [\"a\", \"b\", \"c\"], \"b\": \"b\", \"c\": {\"d\": \"d\"}}, {\"a\": [\"a\"], \"c\": {\"d\": \"d\", \"e\": \"e\"}}]\n",
+    "\n",
+    "report_signature_info(data)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 15,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "/Users/benjamin.wilson/miniconda3/envs/mlflow-dev-env/lib/python3.8/site-packages/_distutils_hack/__init__.py:30: UserWarning: Setuptools is replacing distutils.\n",
+      "  warnings.warn(\"Setuptools is replacing distutils.\")\n"
+     ]
+    },
+    {
+     "data": {
+      "text/html": [
+       "<div>\n",
+       "<style scoped>\n",
+       "    .dataframe tbody tr th:only-of-type {\n",
+       "        vertical-align: middle;\n",
+       "    }\n",
+       "\n",
+       "    .dataframe tbody tr th {\n",
+       "        vertical-align: top;\n",
+       "    }\n",
+       "\n",
+       "    .dataframe thead th {\n",
+       "        text-align: right;\n",
+       "    }\n",
+       "</style>\n",
+       "<table border=\"1\" class=\"dataframe\">\n",
+       "  <thead>\n",
+       "    <tr style=\"text-align: right;\">\n",
+       "      <th></th>\n",
+       "      <th>a</th>\n",
+       "      <th>b</th>\n",
+       "      <th>c</th>\n",
+       "    </tr>\n",
+       "  </thead>\n",
+       "  <tbody>\n",
+       "    <tr>\n",
+       "      <th>0</th>\n",
+       "      <td>[a, b, c]</td>\n",
+       "      <td>b</td>\n",
+       "      <td>{'d': 'd'}</td>\n",
+       "    </tr>\n",
+       "    <tr>\n",
+       "      <th>1</th>\n",
+       "      <td>[a]</td>\n",
+       "      <td>NaN</td>\n",
+       "      <td>{'d': 'd', 'e': 'e'}</td>\n",
+       "    </tr>\n",
+       "  </tbody>\n",
+       "</table>\n",
+       "</div>"
+      ],
+      "text/plain": [
+       "           a    b                     c\n",
+       "0  [a, b, c]    b            {'d': 'd'}\n",
+       "1        [a]  NaN  {'d': 'd', 'e': 'e'}"
+      ]
+     },
+     "execution_count": 15,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "with mlflow.start_run():\n",
+    "    model_info = mlflow.pyfunc.log_model(\n",
+    "        python_model=MyModel(),\n",
+    "        artifact_path=\"test_model\",\n",
+    "        signature=infer_signature(data),\n",
+    "    )\n",
+    "\n",
+    "loaded_model = mlflow.pyfunc.load_model(model_info.model_uri)\n",
+    "prediction = loaded_model.predict(data)\n",
```

Comment:
This can sound like that the users need to do 3 steps to log full signature (1) log with input (2) make prediction with loaded model (3) log with input and prediction, while they actually don't need to. Can we simply make prediction with `prediction = MyModel().predict(input)` and log model only once?

Reply:
Removed those cells and consolidated the signature definition